Sub:add Pilot control page (#6847)
ArduPilot/ardupilot_wiki: sub/source/docs/pilot-control.rst

.. _pilot-control:

=============
Pilot Control
=============

Control of the vehicle can be accomplished in three ways:

- Via Ground Control Stations using:
   - GCS attached Joysticks, or command lines, to send MAVLink `RC override <https://mavlink.io/en/messages/common.html#RC_CHANNELS_OVERRIDE>`__ commands that act as RC inputs (even if no RC receiver is attached)
   - MAVLink ` MANUAL CONTROL <https://mavlink.io/en/messages/common.html#MANUAL_CONTROL>`__ messages which includes axis commands and button states.

- RC control (not enabled by default)

Pilot Control Inputs
====================

> If RC is not enabled, pilot inputs are via the GCS joystick, controlling the vehicle axes, and body-frame movements laterally,forward/back, and up/down in depth. Additionally, lights and camera can be controlled from the GCS.

Reviewer: ```suggestion
If RC is not enabled, pilot inputs are via the GCS joystick, controlling the vehicle's available motion axes, including yaw and attitude rotations, along with body-frame translations laterally, forward/back, and up/down in depth. Additionally, lights and camera can be controlled from the GCS, as well as some generic servo and/or relay outputs.
```

> If RC is enabled, then the pitch/roll/yaw/throttle/lateral/forward inputs are sent from the RC transmitter and mapped from RC channels to these inputs via the ``RCMAP_`` parameters.

Reviewer: `RCMAP` params are [noted as disabled](https://github.com/ArduPilot/ardupilot/blame/407b2707d7618bb0c807385e3fbd9ae98d1cf513/ArduSub/Parameters.h#L235) in the Sub firmware, although I'm unsure whether that's still the case given your recent PR to add RC support.

They're also included in [this blacklist](https://github.com/ArduPilot/ardupilot/blob/407b2707d7618bb0c807385e3fbd9ae98d1cf513/Tools/autotest/param_metadata/mdemit.py#L11), but that's for the markdown version of the parameters docs, and may just be outdated.

If these _do_ work now then [the MAVLink Overrides docs](https://github.com/ArduPilot/ardupilot_wiki/blob/master/sub/source/docs/sub-hardware.rst?plain=1#L23) should be updated to mention/reflect that the motion axis channels can be remapped.

Author: will update once my RCMAP PR is merged

Pilot Control Freedoms by Mode
==============================

> The following assumes the axis is controllable. Some frame configurations do not provide all axes.

Reviewer: ```suggestion
The following assumes each axis is controllable. Some frame configurations do not provide all axes (e.g. a vehicle with only horizontal thrusters may not be able to pitch or roll).
```
Likely worth including an example/explanation around what not providing all axes means / why a frame may not do so. There are already plenty of support questions around using position-required modes for vehicles without a positioning sensor - this seems in a similar vein 😅

> ANG = Angle target. Stick deflection indicates desired axis angle

Reviewer: ```suggestion
ANG = Angle target. Stick deflection indicates desired axis angle.
```

> RAT = Rate target. Sitck deflection indicated desired axis movement/rotation rate.

Reviewer: ```suggestion
RAT = Rate target. Stick deflection indicates desired axis movement/rotation rate.
```

> D = Direct motor output based on stick deflection

Reviewer: ```suggestion
D = Direct motor output based on stick deflection.
```

=========== ===== ===== ==== ===== ======= ======= ======= =========
Mode        Roll  Pitch Yaw  Depth Forward Lateral PosHold DepthHold
=========== ===== ===== ==== ===== ======= ======= ======= =========
MANUAL       D     D     D     D     D       D       No      No
ACRO         RAT   RAT   RAT   RAT   RAT     RAT     No      No
STABILIZE    ANG   ANG   RAT   RAT   RAT     RAT     No      No
ALT_HOLD     ANG   ANG   RAT   RA    RAT     RAT     No      Yes

> POSHOLD      ANG   ANG   RAT   RAT   RAT     RAT     Yes     Yes

Reviewer: `POSHOLD` should potentially be moved one lower, as it introduces additional automation capabilities beyond what `SURFTRAK` provides.

Author: done

> SURFTRAK     ANG   ANG   RAT   RAT   RAT     RAT     No      Yes*

Reviewer: There's a `*` here for `DepthHold`, but no note/explanation further down. Presumably there was an intent to describe that `DepthHold` in `SURFTRAK` mode results in maintaining vertical range from the detected surface, which is sometimes a relative altitude rather than a depth?

Author: yes, added note

> SURFTRAK     ANG   ANG   RAT   RAT   RAT     RAT     Yes     Yes*
> SURFACE      ANG   ANG   RAT   -     RAT     RAT     Yes     No
> CIRCLE       ANG   ANG   RAT   RAT   -       -       Yes     Yes
> MOTORDETECT  -     -     -     -     -       -        -       -

Reviewer 2: ```suggestion
SURFTRAK     ANG   ANG   RAT   RAT   RAT     RAT     No     Yes*
SURFACE      ANG   ANG   RAT   -     RAT     RAT     No     No
CIRCLE       ANG   ANG   RAT   RAT   -       -       Yes     Yes
MOTORDETECT  -     -     -     -     -       -        -       -
```

Author: thanks...
